### ISSUE TITLE
Add `test_that_if_linter()`?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1731]
I often see `{testthat}` tests being skipped using a conditional expression, when `testthat::skip_if_*()` functions are designed exactly for this and far more readable. If there isn't a relevant skip function readily available, it is also easy to create your own (like we do [here](https://github.com/r-lib/lintr/blob/c22485349a2171dddf370e2ebf1f922ea846aa43/tests/testthat/helper.R#L55-L63)). 

We can lint such instances. The warning message can encourage users to check out the existing skip functions and, if needed, to create a custom skip function.

For example,

```r
library(testthat)

if (Sys.getenv("RunAllTests") == "yes") {
  if (requireNamespace("dplyr") && !packageVersion("lme4") >= "1.0") {
    test_that("xyz", {
      ...
    })
  }
}

test_that("xyz", {
  skip_if_not(Sys.getenv("RunAllTests") == "yes")
  skip_if_not_installed("dplyr")
  skip_if(packageVersion("lme4") >= "1.0")
})
```

P.S. `test_that_if_linter()` is just a placeholder; let me know if you have a better name in mind.

**Comment by IndrajeetPatil** [comment]
`testthat_skip_linter()` sounds good to me!

**Comment by AshesITR** [comment]
Good idea!
Are there other testing frameworks with similar mechanics? I'm only familiar with testthat.
